core/sdk/src/main/java/com/mongodb/stitch/core/internal/net/EventStreamReader.java

```diff
@@ -94,6 +94,8 @@ protected final Event processEvent() throws IOException {
         }
         doneOnce = true;
         line = "";
+      } catch (IOException e) {
+        return null;
```

Comment:
[q] why not throw an exception? if we're keeping this, update the comment for this method to explain when this may be null, and add a `@Nullable` annotation